linera-chain/src/chain.rs

```diff
@@ -453,7 +453,7 @@ where
         &mut self,
         block: &Block,
         now: Timestamp,
-    ) -> Result<(Vec<OutgoingMessage>, CryptoHash), ChainError> {
+    ) -> Result<(Vec<OutgoingMessage>, Vec<u32>, CryptoHash), ChainError> {
```

Comment:
nit: Perhaps it's time to define a `struct BlockExecutionOutcome { messages, message_counts, state_hash }`.

Reply:
I also realize now that all the `*CallResult` or `*ExecutionResult` types in `linera-execution` and `linera-sdk` could have been called `*Outcome`.